src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -754,7 +754,23 @@ public function prepare_item_for_response( $item, $request ) {
 			}
 		}
 
-		return $response;
+		/**
+		 * Filters the post data for a REST API response.
+		 *
+		 * The dynamic portion of the hook name, `$this->post_type`, refers to the post type slug.
+		 *
+		 * Possible hook names include:
+		 *
+		 *  - `rest_prepare_wp_template`
+		 *  - `rest_prepare_wp_template_part`
+		 *
+		 * @since 6.6.0
+		 *
+		 * @param WP_REST_Response  $response The response object.
+		 * @param WP_Block_Template $template Template object.
+		 * @param WP_REST_Request   $request Request object.
+		 */
+		return apply_filters( "rest_prepare_{$this->post_type}", $response, $template, $request );
```

Comment:
In our other filter [here](https://github.com/tjcafferkey/wordpress-develop/blob/add-hooked-blocks-metadata-attribute-to-template-response/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L622-L623) instead of documenting it we referenced the WP Posts controller since it is an identical filter.

In this case we can't do that since the second param is different and returns `WP_Block_Template` instead of `WP_Post`.